src/agent/transform-headers-agent.ts

```diff
@@ -39,12 +38,19 @@ class TransformHeadersAgent extends WrappedAgent {
             }
         }
 
+        const typedRequest = request as ClientRequest & {
+            _removedContLen: boolean;
+            _contentLength: number;
+            _removedTE: boolean;
+            agent?: Agent;
+        };
+
         if (!hasConnection) {
-            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || request.agent);
+            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || typedRequest.agent);
             if (shouldSendKeepAlive) {
-                headers.Connection = 'keep-alive';
+                headers['Connection'] = 'keep-alive'; // eslint-disable-line dot-notation
```

Comment:
If headers is typed as a Record, why do you need to use `[]`? What's the error?

Reply:
Because it's a Record, not an interface. The error is that generic properties must be accessed via the index operator.

Reply:
Huh.. I don't have that issue 🤔 
![image](https://user-images.githubusercontent.com/17960496/129043329-8a35e704-3ce4-4b53-802a-8e0e40447ff7.png)

Reply:
Did you try building?

Reply:
VSCode would've errored out if it wasn't valid. I cloned the repo locally and I see what you mean, tho it makes no sense.. You can mark this as resolved

Reply:
Let's keep this open. Some magic behavior happening here that needs to be fixed.

Reply:
Fixed in #39